### PR TITLE
Implement NMF::parameterGradient

### PR DESCRIPTION
**Pull request by jschueller**
needed for otrobopt

### CODE REVIEW

File: lib/src/Base/Func/NumericalMathEvaluationImplementation.cxx [hunk 1]
```diff
@@ -285,7 +285,35 @@ HistoryStrategy NumericalMathEvaluationImplementation::getHistoryOutput() const
...
+  Matrix grad(parameterDimension, outputDimension);
+  for (UnsignedInteger i = 0; i < parameterDimension; ++ i)
+  {
+    for (UnsignedInteger j = 0; j < outputDimension; ++ j)
+    {
+      grad(i, j) = (outS[1 + i][j] - outS[0][j]) / epsilon;
+    }
+  }
+  return grad;
 }
```
Comment (regislebrun) [inline lib/src/Base/Func/NumericalMathEvaluationImplementation.cxx]: To rewrite when operator(const NP &, const NS &) will be available ;-)

**Comment by jschueller** [comment]
@regislebrun, yes, that was the plan

**Comment by regislebrun** [comment]
+1

**Comment by sofianehaddad** [comment]
+1

**Pull request merged**